src/S2EExecutionStateRegisters.cpp

```diff
@@ -168,31 +168,36 @@ bool S2EExecutionStateRegisters::readSymbolicRegion(unsigned offset, void *_buf,
     bool oldAllConcrete = wos->isAllConcrete();
 
     // XXX: deal with alignment and overlaps?
-
-    ref<Expr> value = wos->read(offset, size * 8);
-    uint64_t concreteValue;
-    if (!isa<ConstantExpr>(value)) {
-        if (!concretize) {
-            return false;
+    for (unsigned i = 0; i < size; i += 8) {
+        unsigned csize = (size - i) > 8 ? 8 : (size - i);
+        ref<Expr> value = wos->read(offset + i, csize * 8);
```

Comment:
I think ObjectState::read() can handle sizes > 8 bytes. KLEE has an Expr::Int128 data type.

Reply:
Yes, ObjectState::read() can handle sizes > 8 bytes, but 
concreteValue = m_concretizer->concretize(value, reason.c_str());
is the problem, concreteValue is uint64_t and the following is being called in m_concretizer->concretize:
toConstant(*this, expression, reason.c_str())->getZExtValue()
will assert when constant size > 8 bytes, so I must rewrite the concretize part or reuse the old codes(the easy way)  just like the PR does.

Reply:
ok, sounds good. Please explain this in a comment in the code.